WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -482,6 +519,11 @@ public void onPostSelected(PostModel post) {
     public void onPostButtonClicked(int buttonType, PostModel post) {
         if (!isAdded()) return;
 
+        PostModel existingPost = getPostFromRemoteId(post.getRemotePostId());
+        if (existingPost != null) {
+            post = existingPost;
```

Comment:
The posts in `mSearchResults` don't have local ID's for posts that were already in the store before searching. This will get the store model for us which will have the local ID.